src/library/network_data.cpp

```diff
@@ -111,6 +111,130 @@ std::string Ipv6PrefixToString(ByteArray aPrefix)
     return addr.ToString() + "/" + std::to_string(prefixLength);
 }
 
+XpanId::XpanId(uint64_t val)
+    : mValue(val)
+{
+}
+
+XpanId::XpanId()
+    : XpanId(0)
+{
+}
+
+std::string XpanId::str() const
+{
+    return *this;
+}
+
+bool XpanId::operator==(const XpanId &aOther) const
+{
+    return mValue == aOther.mValue;
+}
+
+bool XpanId::operator!=(const uint64_t aOther) const
+{
+    return !operator==(aOther);
+}
+
+bool XpanId::operator<(const XpanId aOther) const
+{
+    return mValue < aOther.mValue;
+}
+
+XpanId::operator std::string() const
+{
+    std::ostringstream stream;
+    stream << std::setfill('0') << std::setw(sizeof(mValue) * 2) << std::hex << mValue;
+    std::string out = stream.str();
+    std::for_each(out.begin(), out.end(), [](char &c) { c = std::toupper(c); });
+    return out;
+}
+
+/**
+ * Converts hex string to the corresponding integer type.
+ * @attention Makes no validity checks.
+ */
+Error XpanId::FromHex(const std::string &aInput)
+{
+    mValue = 0;
+
+    std::string input = aInput;
+    if (input.substr(0, 2) == "0x")
+    {
+        input = input.substr(2);
+    }
+    if (input.empty() || input.length() > 16)
+        return ERROR_BAD_FORMAT("wrong XPAN string length {}", input.length());
+    for (auto c : input)
+    {
+        if (!std::isxdigit(c))
+        {
+            return ERROR_BAD_FORMAT("not a hex string '{}'", input);
+        }
+    }
+
+    std::istringstream is(input);
+    is >> std::hex >> mValue;
+    return ERROR_NONE;
+}
+
+PanId::PanId(uint16_t aValue)
+    : mValue(aValue)
+{
+}
+
+PanId::PanId()
+    : PanId(0)
+{
+}
+
+PanId &PanId::operator=(uint16_t aValue)
+{
+    mValue = aValue;
+    return *this;
+}
+
+PanId::operator uint16_t() const
+{
+    return mValue;
+}
+
+Error PanId::FromHex(const std::string &aInput)
+{
+    mValue = 0;
+
+    std::string input = aInput;
+    if (input.substr(0, 2) == "0x")
```

Comment:
What about `0X1234`?

Reply:
Why not reuse `Interpreter::ParseInteger` by making it a function in `src/common`?

Reply:
Parsing fixed.